### PR TITLE
EKA-150 | Fixing date checks. Changing logging of datarequest

### PR DESCRIPTION
**Pull request by angshu**

### CODE REVIEW

File: src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs [hunk 1]
```diff
@@ -85,5 +99,18 @@ private List<string> FindPatientData(DataRequest request)
...
+
+        private static void LogDataRequest(DataRequest request)
+        {
+            var ccList = JsonConvert.SerializeObject(request.CareContexts);
+            var requestedHiTypes = string.Join(", ", request.HiType.Select(hiType => hiType.ToString()));
+            var transactionId = request.TransactionId;
+            var from = request.DataRange.From;
+            var to = request.DataRange.To;
+            var callbackUrl = request.CallBackUrl;
+            Log.Information(
```
Comment (ganesanarun) [inline src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs]: Do we need to define these? can't we use them directly?

Comment (angshu) [reply src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs]: Thats what I thought. But it was throwing compilation errors. However, the documentation said such formatting should be fine.

File: src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs [hunk 2]
```diff
@@ -38,17 +41,28 @@ public async Task<Option<Entries>> CollectData(DataRequest dataRequest)
...
         }
 
+        private static DateTime ParseDate(string dateString)
+        {
+            var formatStrings = new string[]
+            {
+                "yyyy-MM-dd", "yyyy-MM-dd hh:mm:ss", "yyyy-MM-dd hh:mm:ss tt", "yyyy-MM-ddTHH:mm:ss.fffzzz", 
+                "dd/MM/yyyy", "dd/MM/yyyy hh:mm:ss", "dd/MM/yyyy hh:mm:ss tt", "dd/MM/yyyyTHH:mm:ss.fffzzz" 
+            };
+            DateTime.TryParseExact(dateString, formatStrings, CultureInfo.CurrentCulture, DateTimeStyles.None,
```
Comment (ganesanarun) [inline src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs]: What I understood is DateTime.Parse or TryParse works pretty well, https://docs.microsoft.com/en-us/dotnet/api/system.datetime.parse?view=netframework-4.8#the-string-to-parse. 

Whatever you have done is OK.

File: src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs (hunk 1 above)
Comment (ganesanarun) [reply src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs]: Not all of them, only ```var transactionId = request.TransactionId;
            var from = request.DataRange.From;
            var to = request.DataRange.To;
            var callbackUrl = request.CallBackUrl```

**Pull request merged**